src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int INITIAL_CAPACITY = 16;
+    public static final double LOAD_FACTOR = 0.75;
+    public static final int GROWTH_FACTOR = 2;
+    private int size;
+    private Node<K, V>[] nodes;
+
+    public MyHashMap() {
+        nodes = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= nodes.length * LOAD_FACTOR) {
+            resize();
+        }
+        putValue(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getHash(key);
+        int position = hash % nodes.length;
```

Comment:
create separate method for that
```suggestion
        int index = getIndex(key);
```